### PR TITLE
perf: rerendering improvements for initial login

### LINKED ISSUE #5448: Optimize wallet connecting interactions  for mobile
### Overview

Enhance rendering performance for low-compute mobile devices, specifically focusing on the initial login and wallet connection phases.

Current performance bottlenecks have been identified in sub-optimal use of React Hooks, notably the absence of `useMemo` and `useCallback` for expensive computations and function references.

### References and additional details

1. Profile the application using Chrome DevTools with 6x throttling to identify components causing re-renders.
2. Implement `useMemo` for computationally expensive render elements.
3. Utilize `useCallback` to memoize function references passed to child components.
4. Review Chakra UI components for potential performance optimizations, such as lazy-loading or conditional rendering.

https://www.notion.so/shapeshift/Mobile-profiling-0321f42373a14e399db7e5255676a97c

### Acceptance Criteria

- Improved in rendering performance on low-compute devices.
- No degradation in user experience or functionality.

### Need By Date

_No response_

### Screenshots/Mockups

_No response_

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Enhances rendering performance for low-compute mobile devices, specifically focusing on the initial login and wallet connection phases. Includes optimizations for other devices.

Current performance bottlenecks have been identified in sub-optimal use of React Hooks, notably the absence of useMemo and useCallback for expensive computations and function references.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #5448

## Testing

Check for regressions on the following (specifically connecting, disconnecting, recovering, backup etc:
- mobile wallet
- native wallet
- metamask
- keepkey

### Engineering

☝️

### Operations

☝️

### Native wallet connecting (with 6x throttling to emulate mobile performance)
before
image

after
image

### CODE REVIEW

File: src/context/WalletProvider/KeepKey/components/Connect.tsx [hunk 1]
```diff
@@ -41,14 +41,16 @@ export const KeepKeyConnect = () => {
   const [loading, setLoading] = useState(false)
   const [error, setError] = useState<string | null>(null)
 
-  // eslint-disable-next-line no-sequences
-  const setErrorLoading = (e: string | null) => (setError(e), setLoading(false))
+  const setErrorLoading = useCallback((e: string | null) => {
+    setError(e)
+    setLoading(false)
+  }, [])
```
Comment (gomesalexandre) [inline src/context/WalletProvider/KeepKey/components/Connect.tsx:47]: 💜

**Review by gomesalexandre — COMMENTED** [review]
Tested locally, functional pass, LGTM minus one possible KK regression:

- Refresh previously connected, locked MM

image

- Refresh connected MM

image

- Connect MM

image

- Native unlock

image


- Switch from MM to native

image

- Native connect

image

- Native refresh

image

- Backup ✅  (no screenshot here sry)
- Create ✅  (no screenshot here sry)

- KeepKey connect

Screenshot 2023-10-16 at 20 39 41
Screenshot 2023-10-16 at 20 40 12

- KeepKey refresh is broken and eventually reverts back to the connect page 🚫 though I'm having some connection issues with that KK, so this may be it locally. Prod was happy when refreshing

image

**Review by gomesalexandre — APPROVED** [review]
Stellar PR ser. Conceptual stamp pending ensuring the spotted KK refresh bug cannot be repro'd, and mobile wallet is tested on an ops build

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5433** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5433" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5433?utm_source=stack-comment).

**Pull request merged**